app/connectors/parsers/HIPAppealParser.scala

```diff
@@ -0,0 +1,114 @@
+import models.appeals.AppealResponseModel
+import play.api.http.Status.{CONFLICT, CREATED}
+import play.api.libs.json.{Format, JsError, JsResult, JsString, JsSuccess, JsValue, Json}
+import uk.gov.hmrc.http.HttpReads.{is4xx, is5xx}
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+import utils.Logger.logger
+import utils.PagerDutyHelper
+import utils.PagerDutyHelper.PagerDutyKeys.{INVALID_JSON_RECEIVED_FROM_1808_API, RECEIVED_4XX_FROM_1808_API, RECEIVED_5XX_FROM_1808_API}
+
+object HIPAppealParser {
+
+  implicit object HIPAppealSubmissionResponseReads extends HttpReads[AppealSubmissionResponse] {
+
+    def read(method: String, url: String, response: HttpResponse): AppealSubmissionResponse = {
+      response.status match {
+        case CREATED =>
+          response.json.validate[AppealResponseModel](AppealResponseModel.format) match {
+            case JsSuccess(model, _) =>
+              Right(model)
+            case _ =>
+              PagerDutyHelper.log("HIPAppealSubmissionResponseReads", INVALID_JSON_RECEIVED_FROM_1808_API)
+              Left(InvalidJson)
+          }
+
+        case CONFLICT =>
+          logger.error(s"[HIPAppealSubmissionResponseReads][read]: Conflict status has been returned returned with body: ${response.body}")
+          Left(DuplicateAppeal)
+
+        case status if is4xx(status) =>
+          PagerDutyHelper.log("HIPAppealSubmissionResponseReads", RECEIVED_4XX_FROM_1808_API)
+          val (status, message) = extractResponse(response)
+          logger.error(s"[HIPAppealSubmissionResponseReads][read]: Unexpected response, status $status returned with reason: $message")
+          Left(BadRequest)
+
+        case status if is5xx(status) =>
+          PagerDutyHelper.log("HIPAppealSubmissionResponseReads", RECEIVED_5XX_FROM_1808_API)
+          val (status, message) = extractResponse(response)
+          Left(UnexpectedFailure(status, s"Unexpected response, status $status returned on submission to HIP with reason:$message"))
+
+        case status =>
+          PagerDutyHelper.logStatusCode("HIPAppealSubmissionResponseReads", status)(RECEIVED_4XX_FROM_1808_API, RECEIVED_5XX_FROM_1808_API)
+          logger.error(s"[HIPAppealSubmissionResponseReads][read]: Unexpected response, status $status returned with body: ${response.body}")
+          Left(UnexpectedFailure(status, s"Unexpected response, status $status returned on submission to HIP"))
+      }
+    }
+
+    private def extractResponse(response: HttpResponse): (Int, String) = {
+      response.json.validate[HIPErrorResponse] match {
+        case JsSuccess(model, _) =>
+          model.failures.headOption match {
+            case Some(failure) =>
+              if(failure.originatedFrom == ETMP) {
+                (failure.dependentSystemHTTPCode, failure.reason)
+              }
+              else {
+                (response.status, failure.reason)
+              }
+            case None => (response.status, response.body)
```

Comment:
We can condense the nested match/if-else to:
`
case Some(failure) if failure.originatedFrom == ETMP => (failure.dependentSystemHTTPCode, failure.reason)
case _ => (response.status, response.body)
`